### PR TITLE
Remove JNR

### PR DESCRIPTION
**Pull request by basil**
Only a handful of plugins use JNR, so it makes more sense as a library plugin than a core dependency. So let's split this out into [`jnr-posix-api-plugin`](https://github.com/jenkinsci/jnr-posix-api-plugin) and remove it from core. I moved [the wrapper class](https://github.com/jenkinsci/jnr-posix-api-plugin/blob/master/src/main/java/hudson/os/PosixAPI.java) there as well as [the sole test](https://github.com/jenkinsci/jnr-posix-api-plugin/blob/master/src/test/java/hudson/os/PosixAPITest.java).

Note that JNR depends on ASM. This PR only removes JNR from the WAR, not ASM. (Removing ASM is more tricky, as core relies on it via Stapler.) Since we weren't declaring an explicit dependency on ASM before, we now declare one in place of a JNR dependency. I verified that this results in the following net change to the WAR:

```
@@ -47,18 +47,11 @@
 WEB-INF/lib/jcommon-1.0.23.jar
 WEB-INF/lib/jenkins-core-2.323-SNAPSHOT.jar
 WEB-INF/lib/jenkins-stapler-support-1.1.jar
-WEB-INF/lib/jffi-1.3.7.jar
-WEB-INF/lib/jffi-1.3.7-native.jar
 WEB-INF/lib/jfreechart-1.0.19.jar
 WEB-INF/lib/j-interop-2.0.8-kohsuke-1.jar
 WEB-INF/lib/j-interopdeps-2.0.8-kohsuke-1.jar
 WEB-INF/lib/jline-2.14.6.jar
 WEB-INF/lib/jna-5.10.0.jar
-WEB-INF/lib/jnr-a64asm-1.0.0.jar
-WEB-INF/lib/jnr-constants-0.10.3.jar
-WEB-INF/lib/jnr-ffi-2.2.9.jar
-WEB-INF/lib/jnr-posix-3.1.12.jar
-WEB-INF/lib/jnr-x86asm-1.0.2.jar
 WEB-INF/lib/json-lib-2.4-jenkins-3.jar
 WEB-INF/lib/jzlib-1.1.3-kohsuke-1.jar
 WEB-INF/lib/kxml2-2.3.0.jar
```

The following plugins are affected:

- [PAM Authentication](https://plugins.jenkins.io/pam-auth/): I opened jenkinsci/pam-auth-plugin#20 to add a dependency on the new `jnr-posix-api-plugin`.
- [CloudBees File Leak Detector](https://plugins.jenkins.io/file-leak-detector/): I opened jenkinsci/file-leak-detector-plugin#10 to add a dependency on the new `jnr-posix-api-plugin`.
- [Maven Repository Scheduled Cleanup](https://plugins.jenkins.io/maven-repo-cleaner/): jenkinsci/maven-repo-cleaner-plugin#4 dropped the dependency on JNR. That PR was merged on June 26, 2018; however, it has not been released.
- [SICCI for Xcode](https://plugins.jenkins.io/sicci_for_xcode/): This plugin is already deprecated and we already dropped support for it in #5560.
- [java.io.tmpdir cleaner](https://plugins.jenkins.io/tmpcleaner/): This plugin is already deprecated and we already dropped support for it in #5560.

The only one of these with significant adoption is PAM Authentication, so from my perspective we can move forward with this PR now that jenkinsci/pam-auth-plugin#20 has been merged and released.

### Testing done

Installed [`jnr-posix-api` 3.1.7-2](https://github.com/jenkinsci/jnr-posix-api-plugin/releases/tag/jnr-posix-api-3.1.7-2) and went to **Manage Jenkins**, **Configure Global Security** and selected **Unix user/group database**. Clicked **Test** and saw **Success**. Clicked **Save**, then logged in as my Unix user with my Unix password successfully. Also tried an incorrect password and verified I was (correctly) rejected.

### Proposed changelog entries

Remove the [Java Native Runtime (JNR)](https://github.com/jnr) library from Jenkins core.

### Proposed upgrade guidelines

The [Java Native Runtime (JNR)](https://github.com/jnr) library has been removed from Jenkins core. Users of [PAM Authentication](https://plugins.jenkins.io/pam-auth/), [CloudBees File Leak Detector](https://plugins.jenkins.io/file-leak-detector/), and [Maven Repository Scheduled Cleanup](https://plugins.jenkins.io/maven-repo-cleaner/) must upgrade those plugins to compatible versions.

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: core/src/main/java/hudson/os/PosixException.java
Comment (jglick) [inline core/src/main/java/hudson/os/PosixException.java:7]: Hmm, still used by `IOUtils.mode`. Wonder if that can be converted to use `IOException`.

**Review by jglick — APPROVED** [review]

File: core/src/main/java/hudson/os/PosixException.java
Comment (basil) [reply core/src/main/java/hudson/os/PosixException.java:7]: Yeah probably.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by usmonster** [comment]
Just for information: Based on some startup issues and logs linked to this, I believe the Datadog plugin also depends on this, so I will open an issue on their side.

**Pull request merged**
